### PR TITLE
Implement dark mode

### PR DESCRIPTION
**Pull request by ThomasMPont**
## Changes

Coloring of UI elements now adapts to the light/dark theme defined by the user's device.

Theme.kt now defines two global themes, light and dark mode, that determine colors for all UI elements.
This change required a total refactor of the way elements are colored:
- UI colors are now defined in Color.kt rather than being listed as resources in colors.xml.
- All existing screens have been refactored to use `MaterialTheme.colorScheme` instead of color resources.

From now on, developers should color elements based on their purpose, e.g. `.border(MaterialTheme.colorScheme.outline)` rather than by picking a fixed color directly.

Here is the palette I have defined for the light theme.
```
primary -> blue
onPrimary -> black
secondary -> red
onSecondary -> black
tertiary -> green
onTertiary -> black
background -> white
onBackground -> black
surface -> darkGray
onSurface -> white
inverseSurface -> blue
inverseOnSurface -> white
error -> red
onError -> white
errorContainer -> lightRed
onErrorContainer -> red
outline -> black
outlineVariant -> blue
```
The dark theme covers the same parameter list.

This palette was chosen to match the current colors of the app as closely as I could. I did sometimes have to change some colors for the naming scheme to stay consistent. Overall, these changes should make colors more consistent between screens.

Dark theme colors are there as placeholders for now, until [the Figma design](https://www.figma.com/design/2kzILCdZRQU1I76wimIaJg/Signify-App?node-id=0-1&node-type=canvas&t=IeHoUpRbBff7gOHc-0) is updated to define some.

## Tests

There isn't too much to test here. I have written some basic regression tests though.

The more important part of testing was going through each screen by hand and making sure their look hasn't changed too much. I haven't linked any screenshots to this description, as the changes affect the entire app.

## Future changes

Originally, this task planned to add an option for the user to override the device's theme, and pick a theme directly through the app's settings. After scouring the internet for hours, I could not find a good way to have the user switch themes inside the app. It turns out this idea is not well documented at all. The only solutions I've found involved a lot of spaghetti code and many new functions.
Thus, after trying to for too long, I have decided not to implement a button to switch themes for now, as this would have lead to a huge PR and a longer task. I'll leave this for another task, and perhaps another developer.

Once dark mode colors are defined on [Figma](https://www.figma.com/design/2kzILCdZRQU1I76wimIaJg/Signify-App?node-id=0-1&node-type=canvas&t=IeHoUpRbBff7gOHc-0), they will need to be pasted into Color.kt.

At the moment, [the Figma design](https://www.figma.com/design/2kzILCdZRQU1I76wimIaJg/Signify-App?node-id=359-1831&node-type=frame&t=IeHoUpRbBff7gOHc-0) is pretty far from the app's actual look. They should both be looked at, and eventually made converge.

### CODE REVIEW

**Comment by ThomasMPont** [comment]
You might have to open details on this SonarCloud analysis, since its numbers got skewed by the broad refactor to all files I've had to make.

**Review by ziyadbagh — DISMISSED** [review]
I ran your branch locally, and I noticed a couple of color inconsistencies:

1. In the quest screen, the titles currently displayed in black should be white:  
   Quest Screen Issue

2. Similarly, on the login screen, the titles shown in black should also be white.
   Login Screen Issue

These adjustments should help improve readability and visual coherence with the app’s theme.

**Comment by ThomasMPont** [comment]
Do you want all text on blue to be white? Or just those lines?

**Comment by ziyadbagh** [comment]
> Do you want all text on blue to be white? Or just those lines?

no the welcome to signify should be white, and the same for the quest titles: Learn letter A/ Learn about letter A
just to match the previous implementation

**Comment by ThomasMPont** [comment]
> no the welcome to signify should be white, and the same for the quest titles: Learn letter A/ Learn about letter A just to match the previous implementation

As we have agreed to on Whatsapp, all text has been changed to be white, except for text on white background.

**Review by ziyadbagh — APPROVED** [review]
Everything looks good to me; I'll approve it.

**Review by GRGool — COMMENTED** [review]
Good work, the dark mode looks great and the code is clean. Small issue I had when running it is with I think the friend list screen which shows up as Dark mode while in light mode. 

https://github.com/user-attachments/assets/b2cfaf01-1def-4b2c-894d-0ed1f321ccf7

**Comment by ThomasMPont** [comment]
> Good work, the dark mode looks great and the code is clean. Small issue I had when running it is with I think the friend list screen which shows up as Dark mode while in light mode.
> 
>  bugmp4.mp4

Make sure you `Request changes` when you point out problems. Otherwise, your comment might go unnoticed.

I'll have it fixed in no time.

**Comment by ThomasMPont** [comment]
Friends list should be fixed now.

File: app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt [hunk 1]
```diff
@@ -29,12 +28,12 @@ fun BottomNavigationMenu(
...
       modifier =
           modifier
-              .border(2.dp, colorResource(R.color.blue)) // Top blue line
+              .border(2.dp, MaterialTheme.colorScheme.outlineVariant) // Top blue line
               .padding(bottom = 2.dp)
               .testTag("BottomNavigationBar")) {
         NavigationBar(
             modifier = Modifier.fillMaxWidth().height(60.dp),
-            containerColor = colorResource(R.color.white) // Set background to white
+            containerColor = MaterialTheme.colorScheme.background // Set background to white
```
Comment (GRGool) [inline app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt:36]: Example of a possibly outdated comment as now the colors are theme dependent.

**Review by GRGool — APPROVED** [review]
Everything looks good to me. Good work on painstakingly going through the colors and refactoring them, this work is gonna help a lot later on. It is well tested and the sonar cloud report shows it. Couple very small points I want to mention : 
- Some comments referring to specific hardcoded colors are still present (e.g., comments mentioning specific colors that are now theme-dependent). These could be removed or updated to avoid confusion. I should have commented one such case in the review.
- The naming follows convention and wasn't very hard to understand, still in my opinion it would be nice to have some comments explaining their purpose (e.g., inverseOnSurface or inverseSurface which I didn't quite get). 
Overall, everything looks great, and it’s well-tested with a strong SonarCloud report. I’ll go ahead and approve.

**Comment by ThomasMPont** [comment]
I'm working on getting rid of such comments. We can merge this later.

**Comment by ThomasMPont** [comment]
I have removed all mentions of specific colors from the screens' code, except for some of the challenge related screens, that are ongoing a refactor anyway.

File: app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt (hunk 1 above)
Comment (ThomasMPont) [reply app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt:36]: For blatant issues like this one, don't hesitate to `Request changes`. I can always dismiss the request if I really don't want to listen anyway.

**Comment by ThomasMPont** [comment]
I had a couple issues with test tags. But everything should be good now.

**Pull request merged**